### DISCUSSION TITLE
Regression in YARP master branch between January 4th and January 13th

### DISCUSSION THREAD

**Discussion by traversaro** [robotology/community#86] [Announcements]
Hi @robotology/all-robotology-developers, 
 a serious regression (a memory leak in the `Bottle` class) was introduced on YARP master branch on January 4th [1] and was discovered and fixed on January 13th [2]. 

If you recently updated an installation based on YARP master branch and your are experiencing any strange behaviour, please update to the latest version of the YARP master branch.

Regression tests using [Valgrind](http://valgrind.org/) have been introduced [3] to make sure that our Continuous Integration system will now catch this kind of issues. 

[1] : https://github.com/robotology/yarp/commit/65847b4cc2867df4a686711233d205837887a8e4
[2] : https://github.com/robotology/yarp/commit/d3846669c1f32fa2a11285a513d85722d11b0b89
[3] : https://github.com/robotology/yarp/pull/667

**Comment by pattacini** [comment]
Great debugging work :+1:
